packages/cdktf/lib/importable-resource.ts

```diff
@@ -0,0 +1,56 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+import { Construct } from "constructs";
+import { TerraformElement } from "./terraform-element";
+import { TerraformProvider } from "./terraform-provider";
+import { ValidateTerraformVersion } from "./validations/validate-terraform-version";
+
+export interface IImportableConfig {
+  terraformResourceType: string;
+  importId: string;
+  provider?: TerraformProvider;
+}
+
+/**
+ * Class used to represent an importable resource.
+ */
+export class ImportableResource extends TerraformElement {
+  constructor(
+    scope: Construct,
+    name: string,
+    private readonly config: IImportableConfig
+  ) {
+    super(scope, name, config.terraformResourceType);
+    this.node.addValidation(
+      new ValidateTerraformVersion(
+        ">=1.5",
+        `Import blocks are only supported for Terraform >=1.5. Please upgrade your Terraform version.`
+      )
+    );
+  }
+
+  /**
+   * Adds this resource to the terraform JSON output.
+   */
```

Comment:
This comment might be a bit misleading if read by a user (i.e. the user might think they need to call this).
```suggestion
  /**
   * Returns the Terraform JSON for an import block for this resource.
   */
```